source/grid/tria_accessor.cc

```diff
@@ -1869,6 +1869,144 @@ CellAccessor<dim, spacedim>::neighbor_of_coarser_neighbor (const unsigned int ne
 
 
 template <int dim, int spacedim>
+bool
+CellAccessor<dim, spacedim>::has_periodic_neighbor (const unsigned int i_face) const
+{
+  typedef TriaIterator<CellAccessor<dim, spacedim>> cell_iterator;
+  /* my_it : is the iterator to the current cell. */
+  cell_iterator my_it(*this);
+  if (this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face))
+      != this->tria->periodic_face_map.end())
+    return true;
+  return false;
+}
+
+
+
+template <int dim, int spacedim>
+TriaIterator<CellAccessor<dim,spacedim> >
+CellAccessor<dim, spacedim>::
+neighbor_or_periodic_neighbor (const unsigned int i_face) const
+{
+  typedef TriaIterator<CellAccessor<dim, spacedim>> cell_iterator;
+  typedef std::pair<cell_iterator, unsigned int> cell_face_pair;
+  /* my_it        : the iterator to the current cell.
+   * my_face_pair : the pair reported by periodic_face_map as its first pair being
+   *                the current cell_face.
+   */
+  cell_iterator my_it(*this);
+  typename std::map<const cell_face_pair, cell_face_pair>::const_iterator my_face_pair =
+    this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face));
+  if (my_face_pair != this->tria->periodic_face_map.end())
+  {
+    /*
+     * Some assertions might be required !
+     */
+    return my_face_pair->second.first;
+  }
+  else
+    return neighbor(i_face);
+}
+
+
+
+template <int dim, int spacedim>
+unsigned int
+CellAccessor<dim, spacedim>::periodic_neighbor_face_no (const unsigned int i_face) const
+{
+  typedef TriaIterator<CellAccessor<dim, spacedim>> cell_iterator;
+  typedef std::pair<cell_iterator, unsigned int> cell_face_pair;
+  /* my_it        : the iterator to the current cell.
+   * my_face_pair : the pair reported by periodic_face_map as its first pair being
+   *                the current cell_face.
+   */
+  cell_iterator my_it(*this);
+  typename std::map<const cell_face_pair, cell_face_pair>::const_iterator my_face_pair =
+    this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face));
+  /*
+   * There should be an assertion, which tells the user that this function should not be
+   * called for a cell which is not located at a periodic boundary !
+   */
+  assert(my_face_pair != this->tria->periodic_face_map.end());
+  return my_face_pair->second.second;
+}
+
```

Comment:
For std::map, that the keys are unique, if we want to just check if a key exists, count() and find() are the same. However, we want to also return the found key. In this case find has the performance advantage, and count will be more readable. I am ok, with either of them, but I like find() better !